packages/generate-persisted-query-manifest/src/index.ts

```diff
@@ -208,17 +254,39 @@ export async function generatePersistedQueryManifest(
     }),
     link: new ApolloLink((operation) => {
       const body = print(sortTopLevelDefinitions(operation.query));
-      manifestOperations.push({
-        id: createHash("sha256").update(body).digest("hex"),
-        name: operation.operationName,
-        type: (
-          operation.query.definitions.find(
-            (d) => d.kind === "OperationDefinition",
-          ) as OperationDefinitionNode
-        ).operation,
-        body,
+      const name = operation.operationName;
+      const type = (
+        operation.query.definitions.find(
+          (d) => d.kind === "OperationDefinition",
+        ) as OperationDefinitionNode
+      ).operation;
+
+      const id = createOperationId(body, {
+        operationName: name,
+        type,
+        createDefaultId() {
+          return defaults.createOperationId(body);
+        },
       });
 
+      // We only need to validate the `id` when using a config file. Without
+      // a config file, our default id function will be used which is
+      // guaranteed to create unique IDs.
+      if (configFile && manifestOperationIds.has(id)) {
+        const message = configFile.message(
+          ERROR_MESSAGES.uniqueOperationId(
+            id,
+            name,
+            manifestOperationIds.get(id)!,
+          ),
+        );
+        message.fatal = true;
```

Comment:
```suggestion
      if (manifestOperationIds.has(id)) {
        addError(
          { file: configFile },
          ERROR_MESSAGES.uniqueOperationId(
            id,
            name,
            manifestOperationIds.get(id)!,
          ),
        );
```

Reply:
Updated in [f5638a1](https://github.com/apollographql/apollo-utils/pull/309/commits/f5638a1beec1ef04530ea6ac4b4cd0d083eed17e)